### PR TITLE
EKF2: robustify terrain validity conditions

### PR DESCRIPTION
**Pull request by bresch**
### Solved Problem
A large negative hagl can lead lead to a valid result as this value is squared to be compared to a variance.

### Solution
Never consider a negative hagl as valid. Also make sure the variance is positive.

### Changelog Entry
For release notes:
```

New parameter: -
Documentation: -
```

### Test coverage
unit tests

### CODE REVIEW

**Comment by MaEtUgR** [comment]
> Never consider a negative hagl as valid.

Until UUV 😈🤣

**Comment by bresch** [comment]
> Until UUV

Terrain could also be the bottom of the ocean :D

File: src/modules/ekf2/EKF/terrain_control.cpp [hunk 1]
```diff
@@ -106,21 +107,29 @@ void Ekf::updateTerrainValidity()
 		float hagl_var = INFINITY;
 		sym::ComputeHaglInnovVar(P, 0.f, &hagl_var);
 
-		if (hagl_var < fmaxf(sq(0.1f * getHagl()), 0.2f)) {
-			valid_hagl_var = true;
+		positive_hagl_var = hagl_var > 0.f;
+
+		if (positive_hagl_var
+		    && (hagl_var < sq(fmaxf(0.1f * getHagl(), 0.5f)))
```
Comment (dagar) [inline src/modules/ekf2/EKF/terrain_control.cpp:113]: Any particular reason for 0.2->0.5? Should it be based on anything or configurable?

**Review by haumarco — APPROVED** [review]

**Review by dagar — APPROVED** [review]

File: src/modules/ekf2/EKF/terrain_control.cpp (hunk 1 above)
Comment (bresch) [reply src/modules/ekf2/EKF/terrain_control.cpp:113]: It's just that we now square it and 0.5^2 ~= 0.2

**Pull request merged**